### PR TITLE
Add the min. changes required to support the GF variant.

### PR DESCRIPTION
**Pull request by PROFESSOR-J-K**

### CODE REVIEW

File: nimble/drivers/dialog_cmac/src/ble_phy.c [hunk 1]
```diff
@@ -1178,6 +1186,8 @@ ble_phy_init(void)
     g_ble_phy_data.phy_whitening = 1;
 #endif
 
+    set_reg32_bits(0x40000904, 0x0004, 1);
```
Comment (andrzej-kaczmarek) [inline nimble/drivers/dialog_cmac/src/ble_phy.c:1189]: move this to ble_rf_init

**Review by andrzej-kaczmarek — CHANGES_REQUESTED** [review]
please add "nimble/phy/cmac:" prefix to commit title

File: nimble/drivers/dialog_cmac/src/ble_phy.c (hunk 1 above)
Comment (PROFESSOR-J-K) [reply nimble/drivers/dialog_cmac/src/ble_phy.c:1189]: This code line deals with suppressing a meta-stability issue from raising an error condition towards the controller, mainly observed in GF variant. Preferably, this should be executed as early as possible and is not related to the RF part. Is there any specific reason why ble_rf_init() is recommended?

Comment (andrzej-kaczmarek) [reply nimble/drivers/dialog_cmac/src/ble_phy.c:1189]: in such case you can put it in `SystemInit` instead: https://github.com/apache/mynewt-core/blob/master/hw/mcu/dialog/cmac/src/system_cmac.c

Comment (PROFESSOR-J-K) [reply nimble/drivers/dialog_cmac/src/ble_phy.c:1189]: Correct. That is the place I was looking for...

Comment (PROFESSOR-J-K) [reply nimble/drivers/dialog_cmac/src/ble_phy.c]: Done. https://github.com/apache/mynewt-core/pull/2992/commits/e110969a71b95f9f89578cdc5fcb2136ba9e17f3

File: nimble/drivers/dialog_cmac/src/ble_rf.c [hunk 2]
```diff
@@ -114,6 +122,20 @@ set_reg16_mask(uint32_t addr, uint16_t mask, uint16_t val)
     *reg = (*reg & (~mask)) | (val & mask);
 }
 
+static inline int read_chip_variant(void)
```
Comment (andrzej-kaczmarek) [inline nimble/drivers/dialog_cmac/src/ble_rf.c:125]:
```
static inline int
read_chip_variant(void)
```

**Comment by andrzej-kaczmarek** [comment]
please rebase to `master` instead or merging `master` branch

**Review by andrzej-kaczmarek — APPROVED** [review]
build failures are not related to this PR, will fix separately

**Pull request merged**